src/components/Button/Button.jsx

```diff
@@ -39,96 +40,99 @@ const Button = forwardRef(({ children, icon, direction, caption, captionColor, d
-        &:hover {
-          fill: ${({ color }) => th.color(`${color}_hover`)};
+const colorVariants = ({ theme: { colors }, color }) => {
+  return variant({
+    prop: 'variant',
+    variants: {
+      filled: {
+        backgroundColor: `${colors[`${color}`]}`,
+        borderColor: `${colors[`${color}`]}`,
+        color: 'white',
+        '&:hover': {
+          backgroundColor: `${colors[`${color}_hover`]}`,
+          borderColor: `${colors[`${color}_hover`]}`
+        },
+        '&:active': {
+          backgroundColor: `${colors[`${color}_active`]}`,
+          borderColor: `${colors[`${color}_active`]}`
+        },
+        '&:disabled': {
+          backgroundColor: `${colors.gray['500']}`,
+          borderColor: `${colors.gray['500']}`,
+          color: `${colors.white}`
         }
-        &:active {
-          fill: ${({ color }) => th.color(`${color}_active`)};
+      },
+      outlined: {
+        backgroundColor: 'transparent',
+        borderColor: `${colors[`${color}`]}`,
+        color: `${colors[`${color}`]}`,
+        '&:hover': {
+          borderColor: `${colors[`${color}_hover`]}`,
+          color: `${colors[`${color}_hover`]}`
+        },
+        '&:active': {
+          borderColor: `${colors[`${color}_active`]}`,
+          color: `${colors[`${color}_active`]}`
+        },
+        '&:disabled': {
+          borderColor: `${colors.gray['500']}`,
+          color: `${colors.gray['500']}`
         }
-        &:disabled {
-          fill: disabled;
+      },
+      text: {
+        backgroundColor: 'transparent',
+        borderColor: 'transparent',
+        color: `${colors[`${color}`]}`,
+        '&:hover': {
+          color: `${colors[`${color}_hover`]}`
+        },
+        '&:active': {
+          color: `${colors[`${color}_active`]}`
+        },
+        '&:disabled': {
+          color: `${colors.gray['500']}`
         }
       }
-      background-color: transparent;
-      border-color: ${({ color }) => th.color(color)};
-      color: ${({ color }) => th.color(color)};
-      &:hover {
-        border-color: ${({ color }) => th.color(`${color}_hover`)};
-        color: ${({ color }) => th.color(`${color}_hover`)};
-      }
-      &:active {
-        border-color: ${({ color }) => th.color(`${color}_active`)};
-        color: ${({ color }) => th.color(`${color}_active`)};
-      }
-      &:disabled {
-        border-color: disabled;
-        color: disabled;
-      }
-    `,
-    text: css`
-      background-color: transparent;
-      border-color: transparent;
-      color: ${({ color }) => th.color(color)};
+    }
+  })
+}
+
+const StyledIcon = styled(Icon)`
+  ${colorVariants}
+`
+
+const Base = styled.button(
+  props => css`
+    cursor: pointer;
+    padding: 4px;
+    border-width: 1px;
+    border-style: solid;
+    border-radius: 4px;
+    min-height: 40px;
+    ${layout}
+    ${space}
+    ${flexbox}
+    ${colorVariants(props)}
+    &:disabled {
+      cursor: initial;
```

Comment:
No `disabled` poderíamos usar `not-allowed`, mas seria legal ver com o pessoal se acham que faz sentido também.

Reply:
Talvez seja legal passar por alguém do design para ver isso. Eu particularmente acho bem feio em alguns SOs aquele blocked.

Reply:
eu curto deixar o nota-allowed

Reply:
Eu discordo, nenhuma lib que conheço utiliza este cursor, é bastante obsoleto. Na web é mais comum vermos indicações visuais menos gritantes para indicar falta de interatividade como cores menos opacas, acredito que manter o cursor como default e setar o pointer-events como none é o melhor caminho.

Reply:
Eu acho o default mais legal também, e a indicação visual ser no componente, modificando a cor. Acho que é o padrão que os designers da Nave aqui seguem em geral. Mas como disse, acho que é válido trocar uma ideia com os padrinhos.

Reply:
Vale o estudo de como outras bibliotecas lidam com isso, boas referências são o material ui e o chakra ui.

Reply:
@devSiso me convenceu. Mas acho que o ideal é a gente consultar os designers mesmo.

Reply:
Acho mais daora o  cursor default também, não vejo o not allowed. Vou de pointer.

Reply:
Fui dar uma olhda, e o do material deixa default, mas o do chakra e ant-design usam o not-allowed. Acho que é uma boa ver com os designers nesse caso

Reply:
Acho que vale o debate com os designers sim, mas levando em consideração outras bibliotecas e cases como argumento para esta conversa afim de chegarmos em um consenso entre boas práticas e comunicação visual.

Reply:
Seguimos assim então pra não travar o PR e levamos a discussão pro design?

Reply:
Opa, massa a discussão! Tava dando uma olhada no que as lib's usam e é bem dividido. De qualquer forma acho que o mais correto é perguntarmos pro design;
|  | usa not-allowed| nao usa n.a  |
|--|--|--|
|ant.design|x| |
|Chakra|x| |
|Material-ui||x|
|grommet||x|
|blueprintjs|x||
|react-bootstrap||x|
|rebassjs||x|


Pessoalmente curto muito os padroes que o Chakra utiliza.

Reply:
Excelente!!!!

Reply:
o resultado da discussão foi esse
![image](https://user-images.githubusercontent.com/45475094/124889704-b9888400-dfad-11eb-93d0-d38cc0f3ea37.png)
levando em consideração o que o paracy falou, vou deixar como arial-disabled e o cursor como not-allowed